neqo-client/src/main.rs

```diff
@@ -678,21 +701,51 @@ mod old {
         }
     }
 
+    impl DownloadNext for PostConnectHandlerOld<'_> {
+        fn download_next(&mut self, client: &mut Connection) -> Res<bool> {
+            let url = self
+                .url_queue
+                .pop()
+                .expect("download_next called with empty queue");
```

Comment:
This will create streams for URLs in LIFO order. If instead of a Vec `url_queue` was a VecDeque we could `push_front` & `pop_front`.